src/Aks/Aks/Utils/ValidateSecureString.cs

```diff
@@ -0,0 +1,40 @@
+﻿using Microsoft.Azure.Commands.Aks.Properties;
+
+using System;
+using System.Management.Automation;
+using System.Runtime.InteropServices;
+using System.Security;
+using System.Text.RegularExpressions;
+
+namespace Microsoft.Azure.Commands.Aks.Utils
+{
+    public sealed class ValidateSecureString: ValidateEnumeratedArgumentsAttribute
+    {
+        public string RegularExpression { get; set; }
+
+        protected override void ValidateElement(object element)
+        {
+            SecureString secureString = element as SecureString;
+            string content = SecureStringToString(secureString);
+            Regex regex = new Regex(RegularExpression);
+            if (!regex.IsMatch(content))
+            {
+                throw new ArgumentException(string.Format(Resources.SecureStringNotValid, RegularExpression));
+            }
+        }
+
+        private string SecureStringToString(SecureString value)
```

Comment:
Please reuse the existing https://github.com/Azure/azure-powershell-common/blob/master/src/Common/Extensions/SecureStringExtensions.cs